File: src/main/scala/com/kirkwoodwest/scalawoods/HardwareBasic.scala
// Written by Kirkwood West - kirkwoodwest.com
// (c) 2024
package com.kirkwoodwest.scalawoods

import com.bitwig.extension.callback.ShortMidiDataReceivedCallback
import com.bitwig.extension.callback.SysexMidiDataReceivedCallback
import com.bitwig.extension.controller.api._

class HardwareBasic(val inputPort: MidiIn,
                    val midiOut: MidiOut) {

  def setInputCallback(midiCallback: ShortMidiDataReceivedCallback): Unit = {
    inputPort.setMidiCallback(midiCallback)
  }

  def setSysexCallback(sysexCallback: SysexMidiDataReceivedCallback): Unit = {
    inputPort.setSysexCallback(sysexCallback)
  }

  def sendMidi(status: Int, data1: Int, data2: Int): Unit = {
    this.midiOut.sendMidi(status, data1, data2)
  }

  def sendSysex(s: String): Unit = {
    this.midiOut.sendSysex(s)
  }
  
}

File: src/main/java/com/kirkwoodwest/openwoods/hardware/HardwareBasic.java
package com.kirkwoodwest.openwoods.hardware;

import com.bitwig.extension.callback.ShortMidiDataReceivedCallback;
import com.bitwig.extension.callback.SysexMidiDataReceivedCallback;
import com.bitwig.extension.controller.api.MidiIn;
import com.bitwig.extension.controller.api.MidiOut;

public class HardwareBasic {
    private final MidiIn midiIn;
    private final MidiOut midiOut;

    public HardwareBasic(MidiIn midiIn, MidiOut midiOut) {
        this.midiIn = midiIn;
        this.midiOut = midiOut;
    }

    public MidiIn getMidiIn() {
        return midiIn;
    }

    public MidiOut getMidiOut() {
        return midiOut;
    }

    public void setInputCallback(ShortMidiDataReceivedCallback midiCallback) {
        midiIn.setMidiCallback(midiCallback);
    }

    public void setSysexCallback(SysexMidiDataReceivedCallback sysexCallback) {
        midiIn.setSysexCallback(sysexCallback);
    }

    public void sendMidi(int status, int data1, int data2) {
        midiOut.sendMidi(status, data1, data2);
    }

    public void sendSysex(String s) {
        midiOut.sendSysex(s);
    }
}
